code/modules/mining/mine_items.dm

```diff
@@ -235,7 +235,7 @@
 	var/obj/item/stack/flag/F = locate() in get_turf(src)
 
 	var/turf/T = get_turf(src)
-	if(!T || !istype(T,/turf/simulated/floor/asteroid))
+	if(!T || !istype(T,/turf/simulated/mineral))
```

Comment:
think better 
if(!T || !istype(T,/turf/simulated/mineral/floor))
or flag can be in wall?